### PR TITLE
filter and search added

### PR DESCRIPTION
**Pull request by BeakalDegefa**
## Changes
_List Changes Introduced by this PR_
1. Product page filter added 
2. search function added

### CODE REVIEW

File: screens/Home/ProductsScreen.tsx [hunk 1]
```diff
@@ -9,11 +9,11 @@ import { Product } from '../../models/products.model';
 
 type ParamList = { Params: { search: string } };
 
-const RenderProducts = (products: Product[]) => {
+const RenderProducts = (products: Product[], search: string) => {
 
   const navigation = useNavigation();
   return (
-    products?.map((product, index) => {
+    products?.filter(productItems => productItems.category === search || productItems.title.toLowerCase().includes(search.toLowerCase()) || productItems.description.toLowerCase().includes(search.toLowerCase()) )?.map((product, index) => {
```
Comment (kmausolf) [inline screens/Home/ProductsScreen.tsx:16]: Nice, I like that you can search on multiple fields at once.

**Review by kmausolf — APPROVED** [review]
Good work on this!

File: screens/Home/ProductsScreen.tsx (hunk 1 above)
Comment (BeakalDegefa) [reply screens/Home/ProductsScreen.tsx:16]: Thank You.

**Review by yelab07 — APPROVED** [review]
Great work Beakal !

**Pull request merged**